types/jest-in-case/jest-in-case-tests.ts

```diff
@@ -41,10 +41,10 @@ test('array', () => {
     expect(global.test).toHaveBeenCalledTimes(3);
     expect(tester).toHaveBeenCalledTimes(3);
 
-    expect(global.describe.mock.calls[0][0]).toBe(title);
-    expect(global.test.mock.calls[0][0]).toBe(testCases[0].name);
-    expect(global.test.mock.calls[1][0]).toBe(testCases[1].name);
-    expect(global.test.mock.calls[2][0]).toBe(testCases[2].name);
+    expect(<string> global.describe.mock.calls[0][0]).toBe(title);
```

Comment:
@jwbay Hmm, it seems in my rush, I removed `toBe()` from the "general matcher", which makes little sense, since we don't want to break existing stuff and people might just be testing `toBe()` to less primitive types as well. 
I'll make sure to add it back.

Reply:
Nothing immediately comes to mind, I'll try to take a deeper look later. I hope you don't mind me marking this PR as needing changes to prevent a merge -- would like this addressed since it's a non-ergonomic breaking change until we figure something out

Reply:
@jwbay Of course!
I was hoping for getting some input on the implementation at the very least, my TypeScript isn't up to par to what I actually want to do here.